src/compatibility.ts

```diff
@@ -74,3 +75,24 @@ export const requiredHubFlagWithDeprecations = requiredHubFlag({
   aliases: ['targetdevhubusername'],
   deprecateAliases: true,
 });
+
+/**
+ * @deprecated
+ */
+export const arrayWithDeprecation = (options: Record<string, unknown> = {}) =>
+  Flags.string({
+    // populate passed options
+    ...options,
+    // overlay those options we want to own
+    deprecated: { message: messages.getMessage('arrayStyleFlagsDeprecated') },
+    multiple: true,
+    parse: async (input: string) => {
+      if (input.includes(',')) {
+        const warningMsg = `${StandardColors.warning(messages.getMessage('warning.prefix'))} ${messages.getMessage(
```

Comment:
Borrowed some code from SfCommand.warn function. Given this will hopefully be short lived, I don't think it is worth the refactor to isolate the formatting.

Reply:
Lifecycle/emitWarning should handle all of that, too.